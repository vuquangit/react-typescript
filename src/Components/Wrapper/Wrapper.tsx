/**
 *
 * Wrapper
 *
 */

import React, { FC } from 'react'
import { WrapperProps } from './Wrapper.types'
import { WrapperComponent } from './Wrapper.styled'
import { ThemeProvider } from 'styled-components'
import * as theme from 'theme/theme'

const Wrapper: FC<WrapperProps> = (props: WrapperProps) => {
  return (
    <ThemeProvider theme={theme}>
      <WrapperComponent {...props} />
    </ThemeProvider>
  )
}

export default Wrapper
