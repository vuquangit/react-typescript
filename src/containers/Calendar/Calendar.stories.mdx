import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'
import Calendar from '.'
// import Space from 'stories/components/styled-system/Space.mdx'
import Styles from 'stories/styles.mdx'
import AppWrapper from '@/test/supports/AppWrapper'

<Meta title="Story/Container/Calendar" component={Calendar} />

<Styles />

# Calendar

With `MDX` we can define a story for `Calendar` right in the middle of our
Markdown documentation.


const TemplateLight: Story = (args) => (
  <AppWrapper themeMode="light">
    <Calendar {...args} />
  </AppWrapper>
)

const TemplateDark: Story = (args) => (
  <AppWrapper themeMode="dark">
    <Calendar {...args} />
  </AppWrapper>
)

<Canvas>
  <Story
    name="Light"
  >
    {TemplateLight.bind({})}
  </Story>
  <Story
    name="Dark"
  >
    {TemplateDark.bind({})}
  </Story>
</Canvas>

