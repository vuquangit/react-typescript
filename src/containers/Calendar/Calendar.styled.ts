import styled from 'styled-components'
// import { prop } from 'styled-tools'

// import { ILoaderProps } from './Loader.types'

export const CalendarWrapper = styled('div')`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`

export const CalendarContentWrapper = styled('div')`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  justify-content: space-between;
  width: 100%;
  height: 100%;
`

export const CalendarDayNames = styled('div')`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 100%;
`

export const CalendarHeader = styled('div')`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  padding: 8px;
`

export const CalendarBody = styled('div')`
  border-top: 0.05rem solid #e3e3e7;
  border-left: 0.05rem solid #e3e3e7;
  width: 100%;
`

export const CalendarMonthList = styled('div')`
  display: flex;
  align-items: center;
  flex-direction: row;
  flex-wrap: wrap;
  width: 100%;
  border-top: 0.05rem solid #e3e3e7;
  border-left: 0.05rem solid #e3e3e7;
`

export const CalendarMonthListItem = styled('div')`
  border-bottom: 0.05rem solid #e3e3e7;
  border-right: 0.05rem solid #e3e3e7;
  width: calc(100% * 4 / 12);
  padding: 1rem;
  height: 7rem;
  cursor: pointer;

  &:hover {
    background-color: #5f5f5f;
  }
`
