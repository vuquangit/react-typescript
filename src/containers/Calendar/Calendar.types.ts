export declare type IDate = {
  date: number
  dayName: string
  iso: string
  type?: string
}

export type TWeek = IDate[]

export declare type TWeeks = TWeek[]

export const CALENDAR_TYPE_YEARS = 'years'
export const CALENDAR_TYPE_YEAR = 'year'
export const CALENDAR_TYPE_MONTH = 'month'
export const CALENDAR_TYPE_DAYS = 'days'

export const ON_TYPE_PREV_DATE_MORE = 'prevDateMore'
export const ON_TYPE_PREV_DATE = 'prevDate'
export const ON_TYPE_TODAY = 'today'
export const ON_TYPE_NEXT_DATE = 'nextDate'
export const ON_TYPE_NEXT_DATE_MORE = 'nextDateMOre'

export type TCalendarType =
  | typeof CALENDAR_TYPE_YEARS
  | typeof CALENDAR_TYPE_YEAR
  | typeof CALENDAR_TYPE_MONTH
  | typeof CALENDAR_TYPE_DAYS

export type TChangeTimeType =
  | typeof ON_TYPE_PREV_DATE_MORE
  | typeof ON_TYPE_PREV_DATE
  | typeof ON_TYPE_TODAY
  | typeof ON_TYPE_NEXT_DATE
  | typeof ON_TYPE_NEXT_DATE_MORE
