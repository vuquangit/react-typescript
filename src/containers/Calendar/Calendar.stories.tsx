import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import AppWrapper from '@/test/supports/AppWrapper'
import Calendar from '.'
// import {} from './Calendar.types'

export default {
  title: 'Story/Container/Calendar',
  component: Calendar,
  argTypes: {},
} as Meta

const TemplateLight: Story = (args) => (
  <AppWrapper themeMode="light">
    <Calendar {...args} />
  </AppWrapper>
)

export const Light = TemplateLight.bind({})

const TemplateDark: Story = (args) => (
  <AppWrapper themeMode="dark">
    <Calendar {...args} />
  </AppWrapper>
)

export const Dark = TemplateDark.bind({})
