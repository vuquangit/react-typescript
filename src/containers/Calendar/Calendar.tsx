import React, { FC, useState, useEffect, useCallback } from 'react'
import { getMatrix, dayNames } from 'utils/date'
import { TWeeks } from './Calendar.types'
import CalendarDetail from 'components/CalendarDetail/CalendarDetail'
import Button from 'components/Button'
import {
  CalendarWrapper,
  CalendarContentWrapper,
  CalendarDayNames,
  CalendarHeader,
  CalendarBody,
  CalendarMonthList,
  CalendarMonthListItem,
} from './Calendar.styled'

const Calendar: FC = () => {
  // Note: month start from 0 to 11
  const [time, setTime] = useState<Date>(new Date())
  const [weeks, setWeeks] = useState<TWeeks>()
  const [calendarType, setCalendarType] = useState<
    'years' | 'year' | 'month' | 'days'
  >('days')
  const [yearStart, setYearStart] = useState<number>(0)

  const getWeeks = async (date: Date) => {
    const matrix = await getMatrix(date)
    setWeeks(matrix)
  }

  const onChangeTime = useCallback(
    (type: 'prevYear' | 'prevMonth' | 'today' | 'nextMonth' | 'nextYear') => {
      const val = new Date()

      let numberChange = 1
      if (
        (calendarType === 'month' &&
          (type === 'prevYear' || type === 'nextYear')) ||
        ((calendarType === 'year' || calendarType === 'years') &&
          (type === 'prevMonth' || type === 'nextMonth'))
      )
        numberChange = 10
      else if (
        (calendarType === 'year' || calendarType === 'years') &&
        (type === 'prevYear' || type === 'nextYear')
      )
        numberChange = 100

      if (
        type === 'prevYear' ||
        (type === 'prevMonth' && calendarType !== 'days')
      ) {
        val.setFullYear(
          time.getFullYear() - numberChange,
          time.getMonth(),
          time.getDate()
        )
      } else if (type === 'prevMonth' && calendarType === 'days') {
        val.setFullYear(time.getFullYear(), time.getMonth() - 1, time.getDate())
      } else if (type === 'nextMonth' && calendarType === 'days') {
        val.setFullYear(time.getFullYear(), time.getMonth() + 1, time.getDate())
      } else if (
        type === 'nextYear' ||
        (type === 'nextMonth' && calendarType !== 'days')
      ) {
        val.setFullYear(
          time.getFullYear() + numberChange,
          time.getMonth(),
          time.getDate()
        )
      }

      setTime(val)

      if (type === 'today') setCalendarType('days')
    },
    [calendarType, time]
  )

  const onSelectMonth = (month: number) => {
    const val = new Date(time)
    val.setFullYear(time.getFullYear(), month, time.getDate())
    setTime(val)
    setCalendarType('days')
  }

  const onSelectYear = (year: number) => {
    const val = new Date(time)
    val.setFullYear(year, time.getMonth(), time.getDate())
    setTime(val)
    setCalendarType('month')
  }

  const onSelectYears = (years: number) => {
    const val = new Date(time)
    val.setFullYear(years, time.getMonth(), time.getDate())
    setTime(val)
    setCalendarType('year')
  }

  const onDateHeaderClick = useCallback(() => {
    if (calendarType === 'days') setCalendarType('month')
    else if (calendarType === 'month') setCalendarType('year')
    else if (calendarType === 'year') setCalendarType('years')
  }, [calendarType])

  const getYearStart = useCallback(() => {
    const yearHasSelected = time.getFullYear() + ''

    if (calendarType === 'year') {
      const year2Fill = parseInt(
        yearHasSelected.split('').slice(0, -1).concat('1').join('')
      )
      const _yearStart = yearHasSelected.endsWith('0')
        ? year2Fill - 10
        : year2Fill

      setYearStart(_yearStart)
    } else if (calendarType === 'years') {
      const year2Fill = parseInt(
        yearHasSelected.split('').slice(0, -2).concat('01').join('')
      )
      const _yearStart = yearHasSelected.endsWith('0')
        ? year2Fill - 100
        : year2Fill

      setYearStart(_yearStart)
    }
  }, [calendarType, time])

  useEffect(() => {
    getYearStart()
    calendarType === 'days' && getWeeks(time)
  }, [calendarType, getYearStart, time])

  const renderCalendarContent = useCallback(() => {
    return weeks?.map((week, index) => {
      return (
        <CalendarContentWrapper key={index}>
          {week.map((item) => (
            <CalendarDetail
              key={item.iso}
              {...item}
              maxWidth={100 / week.length}
            />
          ))}
        </CalendarContentWrapper>
      )
    })
  }, [weeks])

  const renderDayNames = useCallback(
    () => dayNames.map((item) => <div key={item}>{item}</div>),
    []
  )

  const renderMonthList = () =>
    Array(12)
      .fill(0)
      .map((_, index) => (
        <CalendarMonthListItem key={index} onClick={() => onSelectMonth(index)}>
          {index + 1}
        </CalendarMonthListItem>
      ))

  const renderYearList = () =>
    Array(10)
      .fill(yearStart)
      .map((item, index) => {
        const year = item + index

        return (
          <CalendarMonthListItem key={index} onClick={() => onSelectYear(year)}>
            {year}
          </CalendarMonthListItem>
        )
      })

  const renderYearsList = () =>
    Array(10)
      .fill(yearStart)
      .map((item, index) => {
        const years = item + index * 100

        return (
          <CalendarMonthListItem
            key={index}
            onClick={() => onSelectYears(years)}
          >
            {`${years}-${years + 99} `}
          </CalendarMonthListItem>
        )
      })

  return (
    <CalendarWrapper>
      <CalendarHeader>
        <div>
          <Button
            fontSize={[18, 24]}
            border={0}
            cursor="pointer"
            onClick={onDateHeaderClick}
            disabled={calendarType === 'years'}
          >
            {calendarType === 'days' &&
              time.getMonth() + 1 + `/` + time.getFullYear()}
            {calendarType === 'month' && time.getFullYear()}
            {calendarType === 'year' && `${yearStart} - ${yearStart + 9}`}
            {calendarType === 'years' && `${yearStart} - ${yearStart + 99}`}
          </Button>
        </div>
        <div>
          <Button
            onClick={() => onChangeTime('prevYear')}
            padding="8px"
            cursor="pointer"
            disabled={time.getFullYear() <= 200}
          >{`<<`}</Button>
          <Button
            onClick={() => onChangeTime('prevMonth')}
            padding="8px"
            cursor="pointer"
            hidden={calendarType === 'years'}
          >{`<`}</Button>
          <Button
            onClick={() => onChangeTime('today')}
            padding="8px"
            cursor="pointer"
          >
            Today
          </Button>
          <Button
            onClick={() => onChangeTime('nextMonth')}
            padding="8px"
            cursor="pointer"
            hidden={calendarType === 'years'}
          >{`>`}</Button>
          <Button
            onClick={() => onChangeTime('nextYear')}
            padding="8px"
            cursor="pointer"
          >{`>>`}</Button>
        </div>
      </CalendarHeader>

      {calendarType === 'days' && (
        <>
          <CalendarDayNames>{renderDayNames()}</CalendarDayNames>
          <CalendarBody>{renderCalendarContent()}</CalendarBody>
        </>
      )}

      {calendarType === 'month' && (
        <CalendarMonthList>{renderMonthList()}</CalendarMonthList>
      )}

      {calendarType === 'year' && (
        <CalendarMonthList>{renderYearList()}</CalendarMonthList>
      )}
      {calendarType === 'years' && (
        <CalendarMonthList>{renderYearsList()}</CalendarMonthList>
      )}
    </CalendarWrapper>
  )
}

export default Calendar
