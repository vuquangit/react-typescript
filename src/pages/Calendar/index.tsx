import React, { FC } from 'react'

import { DefaultLayout } from 'layouts'
import Container from 'components/Container'
import Calendar from 'components/Calendar'

const CalendarPage: FC = () => {
  return (
    <DefaultLayout>
      <Container>
        <Calendar />
      </Container>
    </DefaultLayout>
  )
}

export default CalendarPage
