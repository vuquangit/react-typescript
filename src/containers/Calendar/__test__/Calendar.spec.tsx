/**
 *
 * Tests for Calendar
 *
 */

import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import Calendar from '..'

import Wrapper from '@/test/supports/AppWrapper'

let container: any = null
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  // cleanup on exiting
  document.body.removeChild(container)
  container = null
})

describe('App Calendar', () => {
  it('Test get default Calendar', async () => {
    // Test first render and componentDidMount
    await act(async () => {
      ReactDOM.render(
        <Wrapper themeMode="light">
          <Calendar />
        </Wrapper>,
        container
      )
    })

    expect(container).toMatchSnapshot()
  })
})
