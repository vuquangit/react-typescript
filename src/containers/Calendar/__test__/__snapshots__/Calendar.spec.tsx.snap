// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Calendar Test get default Calendar 1`] = `
.c2 {
  border: 0;
  font-size: 18px;
  cursor: pointer;
  border: 0;
  background-color: #000;
  color: #fff;
  font-size: 16px;
}

.c3 {
  padding: 8px;
  cursor: pointer;
  border: 0;
  background-color: #000;
  color: #fff;
  font-size: 16px;
}

.c8 {
  cursor: default;
  border: 0;
}

.c7 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 7rem;
  border-bottom: 0.05rem solid #e3e3e7;
  border-right: 0.05rem solid #e3e3e7;
  cursor: default;
  pointer-events: none;
  max-width: 14.285714285714286%;
}

.c7:after {
  background: rgba(46,62,72,0.9);
  border-radius: 0.15rem;
  bottom: 100%;
  color: #fff;
  content: attr(data-tooltip);
  display: block;
  font-size: 0.8rem;
  left: 50%;
  max-width: 320px;
  opacity: 0;
  overflow: hidden;
  padding: 0.2rem 0.4rem;
  pointer-events: none;
  position: absolute;
  text-overflow: ellipsis;
  -webkit-transform: translate(-50%,0.4rem);
  -ms-transform: translate(-50%,0.4rem);
  transform: translate(-50%,0.4rem);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  white-space: pre;
  z-index: 300;
}

.c7:focus:after,
.c7:hover:after {
  opacity: 1;
  -webkit-transform: translate(-50%,-0.2rem);
  -ms-transform: translate(-50%,-0.2rem);
  transform: translate(-50%,-0.2rem);
}

.c10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  height: 7rem;
  border-bottom: 0.05rem solid #e3e3e7;
  border-right: 0.05rem solid #e3e3e7;
  cursor: pointer;
  pointer-events: auto;
  max-width: 14.285714285714286%;
}

.c10:after {
  background: rgba(46,62,72,0.9);
  border-radius: 0.15rem;
  bottom: 100%;
  color: #fff;
  content: attr(data-tooltip);
  display: block;
  font-size: 0.8rem;
  left: 50%;
  max-width: 320px;
  opacity: 0;
  overflow: hidden;
  padding: 0.2rem 0.4rem;
  pointer-events: none;
  position: absolute;
  text-overflow: ellipsis;
  -webkit-transform: translate(-50%,0.4rem);
  -ms-transform: translate(-50%,0.4rem);
  transform: translate(-50%,0.4rem);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  white-space: pre;
  z-index: 300;
}

.c10:focus:after,
.c10:hover:after {
  opacity: 1;
  -webkit-transform: translate(-50%,-0.2rem);
  -ms-transform: translate(-50%,-0.2rem);
  transform: translate(-50%,-0.2rem);
}

.c9 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  height: 1.4rem;
  margin-right: 0.2rem;
  margin-top: 0.2rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0.05rem solid transparent;
  border-radius: 50%;
  color: #6c777e;
  cursor: pointer;
  font-size: 0.8rem;
  line-height: 1rem;
  outline: none;
  padding: 0.15rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 1.4rem;
  border-color: none;
  color: inherit;
}

.c13 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  height: 1.4rem;
  margin-right: 0.2rem;
  margin-top: 0.2rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0.05rem solid transparent;
  border-radius: 50%;
  color: #6c777e;
  cursor: pointer;
  font-size: 0.8rem;
  line-height: 1rem;
  outline: none;
  padding: 0.15rem;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 1.4rem;
  border-color: #cde4ff;
  color: #439aff;
}

.c11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  line-height: 1;
  overflow-y: auto;
  padding: 0.2rem;
}

.c12 {
  border-radius: 0.15rem;
  font-size: 0.7rem;
  display: block;
  margin: 0.15rem auto;
  overflow: hidden;
  padding: 3px 4px;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #10b742;
  background: #d3f1dc;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  padding: 8px;
}

.c5 {
  border-top: 0.05rem solid #e3e3e7;
  border-left: 0.05rem solid #e3e3e7;
  width: 100%;
}

@media screen and (min-width:40em) {
  .c2 {
    font-size: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <button
          class="c2"
          cursor="pointer"
          data-testid="app-button"
          font-size="18,24"
        >
          12/2020
        </button>
      </div>
      <div>
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
        >
          &lt;&lt;
        </button>
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
        >
          &lt;
        </button>
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
        >
          Today
        </button>
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
        >
          &gt;
        </button>
        <button
          class="c3"
          cursor="pointer"
          data-testid="app-button"
        >
          &gt;&gt;
        </button>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        Sun
      </div>
      <div>
        Mon
      </div>
      <div>
        Tue
      </div>
      <div>
        Wed
      </div>
      <div>
        Thu
      </div>
      <div>
        Fri
      </div>
      <div>
        Sat
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            29
          </button>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            30
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            1
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            2
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            3
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            4
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            5
          </button>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            6
          </button>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            7
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            8
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            9
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            10
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            11
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            12
          </button>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            13
          </button>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c13"
          >
            14
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            15
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            16
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            17
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            18
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            19
          </button>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            20
          </button>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            21
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            22
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            23
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            24
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            25
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            26
          </button>
        </div>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            27
          </button>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            28
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            29
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            30
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            31
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c10"
          data-tooltip="In: 08:00 - Out: 17:37"
        >
          <button
            class="c8 c9"
          >
            1
          </button>
          <div
            class="c11"
          >
            <span
              class="c12"
            >
              Attend OK
            </span>
            <span
              class="c12"
            >
              Leave OK
            </span>
          </div>
        </div>
        <div
          class="c7"
          data-tooltip="true"
        >
          <button
            class="c8 c9"
          >
            2
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
